jax/_src/nn/functions.py

```diff
@@ -517,6 +518,37 @@ def glu(x: ArrayLike, axis: int = -1) -> Array:
 logsumexp = _logsumexp
 
 
+@partial(api.jit, static_argnames=("axis", "keepdims"))
+def logmeanexp(
+    x: ArrayLike,
+    axis: int | tuple[int, ...] | None = -1,
+    where: ArrayLike | None = None,
+    keepdims: bool = False,
+) -> Array:
+  r"""Log mean exp.
+
+  Computes the function:
+
+  .. math::
+    \text{logmeanexp}(x) = \log \frac{1}{n} \sum_{i=1}^n \exp x_i = \text{logsumexp}(x) - \log n
+
+  Args:
+    x: Input array.
+    axis: Axis or axes along which to reduce.
+    where: Elements to include in the reduction. Optional.
+    keepdims: Preserve the dimensions of the input.
+
+  Returns:
+    An array.
```

Comment:
`logsumexp` provides an option to return the sign separately, for cases when the sum is negative. Should `logmeanexp` provide similar functionality?